.mvn/wrapper/MavenWrapperDownloader.java

```diff
@@ -108,10 +108,12 @@ protected PasswordAuthentication getPasswordAuthentication() {
         URL website = new URL(urlString);
         ReadableByteChannel rbc;
         rbc = Channels.newChannel(website.openStream());
-        FileOutputStream fos = new FileOutputStream(destination);
-        fos.getChannel().transferFrom(rbc, 0, Long.MAX_VALUE);
-        fos.close();
-        rbc.close();
+        try (FileOutputStream fos = new FileOutputStream(destination);) {
+            fos.getChannel().transferFrom(rbc, 0, Long.MAX_VALUE);
+        } finally {
+            fos.close();
+            rbc.close();
+        }
```

Comment:
The idea of the `try-with-resources` block is that the `close` methods are not executed manually by developers. The generated bytecode does it. The proposed change executes the `fos#close` method twice.

The `rbc` object could be also added to the `try-with-resources` block.

This:
```java
try (ReadableByteChannel rbc = Channels.newChannel(website.openStream()); FileOutputStream fos = new FileOutputStream(destination)) {
    fos.getChannel().transferFrom(rbc, 0, Long.MAX_VALUE);
}
```
replaces this:
```java
ReadableByteChannel rbc;
rbc = Channels.newChannel(website.openStream());
FileOutputStream fos = new FileOutputStream(destination);
fos.getChannel().transferFrom(rbc, 0, Long.MAX_VALUE);
fos.close();
rbc.close();
```